ssl/statem/statem_srvr.c

```diff
@@ -527,7 +532,9 @@ static WRITE_TRAN ossl_statem_server13_write_transition(SSL *s)
          * Following an initial handshake we send the number of tickets we have
          * been configured for.
          */
-        if (s->hit || s->num_tickets <= s->sent_tickets) {
+        if (s->ext.extra_tickets_expected > 0) {
+            s->ext.extra_tickets_expected--;
```

Comment:
Would conditioning on having completed the handshake make sense?

Reply:
That would probably make sense, yes.  Thanks!

Reply:
Unfortunately, neither `SSL_in_init()` nor `SSL_is_init_finished()` suffice for this call site (at least in my current model for the API), since in order to write new handshake messages (tickets) we go back "in init", and for the case when we want to write more than one ticket in a row, we have to *stay there* for multiple passes through this logic.  So when I tried it, the "send two tickets in a row" test only sent one ticket.

Reply:
`SSL_IS_FIRST_HANDSHAKE` might do what you need to figure out whether the original handshake has completed or not

Reply:
I added the check here, though later testing indicates that in a normal 1.3 handshake `SSL_IS_FIRST_HANDSHAKE()` will return 0 by the time we are writing tickets.  (Which makes sense, since it's looking at finished lengths, and NST happens after Finished.)